### ISSUE TITLE
Support for more complicated server params

### ISSUE THREAD

**Issue by djoodle** [movio/bramble#50]
Hi,

We use envoy in our service mesh - so to communicate to any of the upstream services we do it via the same URL and modify a header (to identify the targeted cluster). Seems like supporting more custom configurations per service could be useful (e.g. adding headers), but couldn't see how I cold configure that with bramble.

Is this possible/do you have an example?

If not could is this something you'd consider as a feature request?

Cheers!

**Comment by sodabrew** [comment]
Some services I've worked with have their introspection queries behind API keys, so this would be a useful feature for me as well. Just to test bramble, I hardcoded the extra headers I needed, but before going further I'd want to take the time to break this out into a config setting.

Some questions on use cases and scoping:
* Would a static set of Header - Value pairs for each server work?
* Would there need to be an external login flow like for JWT or OAuth?
* If there's any complicated logic, should that be a plugin written in Go or a generic "execute a script" flow?
